### PR TITLE
native: negative integer literals

### PR DESCRIPTION
**Pull request by Eliyaan**
Use 64 bits instruction for all the mov64 registers and add the same function for unsigned ints.
Support negative integer literals.
Throw an native error when printing unsigned integers as it is not supported.
Add tests for negative integer literals.

### CODE REVIEW

File: vlib/v/gen/native/amd64.v [hunk 1]
```diff
@@ -532,37 +532,66 @@ fn (mut c Amd64) mov32(reg Amd64Register, val i32) {
...
+		else {
+			eprintln('unhandled mov64u ${reg}')
+		}
+	}
+	c.g.write64(val)
+	c.g.println('mov64 ${reg}, ${val}')
+}
+
+fn (mut c Amd64) mov64(reg Register, val i64) {
+	// see AMD64 Architecture Programmer's Manual Volume 3 about the MOV instruction to have a detailed explaination
```
Comment (spytheman) [inline vlib/v/gen/native/amd64.v:571]:
```suggestion
	// see AMD64 Architecture Programmer's Manual Volume 3 about the MOV instruction to have a detailed explanation
```

File: vlib/v/gen/native/tests/for.vv [hunk 2]
```diff
@@ -33,6 +33,9 @@ fn simple_for_test() {
 		println('loop7')
 		i++
 	}
+	for j in -2 .. 3 {
+		println('loop8')
```
Comment (spytheman) [inline vlib/v/gen/native/tests/for.vv:37]: can it print `j` in the loop too?

**Review by spytheman — APPROVED** [review]
Excellent work.

**Comment by medvednikov** [comment]
Wonderful!
